### PR TITLE
Health endpoints

### PR DESCRIPTION
**Pull request by JohannesFinsveen**
See HealthEndpoints.md

### CODE REVIEW

**Comment by JohannesFinsveen** [comment]
Very much agree with the EnterMaintanceMode and ExitMaintanceMode , Shutdown seems dramatic :-)   

On alive: Very much agree it should just return 200 if it cant read the file.  
It seems from the nais docs, they prefer 2 endpoints. https://doc.nais.io/workloads/explanations/good-practices/#implements-readiness-and-liveness-endpoints
I feel I need some easy way to get which tags where used to make the running build, but it does not have to be called alive :)     
Small metting, perhaps ..

**Comment by JohannesFinsveen** [comment]
The  alive-endpoint: Do we wish to move the "build info" to another endpoint (like ping)?
The  2 MaintanceMode-endpoints: Do we wish to merge them into 1?

**Comment by runejo** [comment]
> The 2 MaintanceMode-endpoints: Do we wish to merge them into 1?

I like the idea of just one endpoint. I stole this from https://www.splinter.dev/community/planning/rest_api_maintenance_mode.html 
`GET /admin/maintenance` for checking if maintenance mode is enabled
`POST /admin/maintenance` for enabling/disabling maintenance mode

File: HealthEndpoints.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+/health/ready
+/health/alive
+```
+
+Their usefullness depend on how Pxwebapi is installed. 
+If you have a single pxwebapi instance and install it as a package,
+ you will not find them very usefull. 
+
+Their intended use is if you run Pxwebapi as Docker in a kubernetes cluster or 
+have more than one server behind a loadbalancer.
```
Comment (runejo) [inline HealthEndpoints.md:15]: I think maintenance mode is not for Kubernetes deployments,  but it's useful when you have dedicated serveres behind a loadbalancer. Like we do, on-prem :-)

Comment (JohannesFinsveen) [reply HealthEndpoints.md:15]: I "googled with AI" They suggest PUT with true false in body ?  
The get would be for "internal use" to see why health/ready is not 200 ?

**Comment by JohannesFinsveen** [comment]
There was a more buildin way to da this, see the jira issue

**Pull request closed without merging**